### PR TITLE
Add contextual tallies to filters

### LINKED ISSUE #671: Teammate autocomplete instantly fills to "None" if you delete the existing character name
Steps to reproduce:
1. Set a teammate
2. Click the field
3. Backspace to delete the name
4. Once the name is completely deleted, the field will close, and instantly select "None", preventing the user from typing a different name

Pretty sure this happens with all autocompletes, but haven't thoroughly tested

### LINKED ISSUE #885: Add (bright) indication if no Artifact Set Conditionals have been selected
image

### LINKED ISSUE #886: Add indication for how many artifacts have been excluded by the  Artifact Level Filter
image

### PR DESCRIPTION
**Pull request by frzyc**
Add counters to applicable filters.
image
image
Resolve #886  
Resolve #671  
Resolve #885

### CODE REVIEW

File: apps/frontend/src/app/Components/GeneralAutocomplete.tsx [hunk 1]
```diff
@@ -17,22 +30,81 @@ export default function GeneralAutocomplete<T extends string>({ options, valueKe
       return newValue !== null && onChange(newValue.key)
     }}
     isOptionEqualToValue={(option, value) => option.key === value.key}
-    renderInput={props => <SolidColoredTextField
```
Comment (nguyentvan7) [inline apps/frontend/src/app/Components/GeneralAutocomplete.tsx:20]: We lose the colored background that is throughout the site with this change. Did you intend this?

Comment (frzyc) [reply apps/frontend/src/app/Components/GeneralAutocomplete.tsx:20]: yes the colored field background does not look good in autocompletes, I found.

Comment (frzyc) [reply apps/frontend/src/app/Components/GeneralAutocomplete.tsx:20]: might need to look at how autocomplete looks with ArtifactCard

**Comment by frzyc** [comment]
should see if #671 can be resolved as well

**Comment by frzyc** [comment]
impl #424 as well

File: apps/frontend/src/app/Components/GeneralAutocomplete.tsx [hunk 2]
```diff
@@ -1,38 +1,101 @@
...
     value={value}
-    clearIcon={key !== clearKey ? undefined : null}
     onChange={(event, newValue, reason) => {
-      if (reason === "clear" && clearKey !== undefined) return onChange(clearKey)
-      return newValue !== null && onChange(newValue.key)
+      if (reason === "clear") return onChange([])
+      return newValue !== null && onChange(newValue.map(v => v.key))
     }}
     isOptionEqualToValue={(option, value) => option.key === value.key}
-    renderInput={props => <SolidColoredTextField
```
Comment (nguyentvan7) [inline apps/frontend/src/app/Components/GeneralAutocomplete.tsx:20]: With this, we can delete the SolidColoredTextField component

File: apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/MainStatSelectionCard.tsx [hunk 3]
```diff
@@ -9,47 +8,91 @@ import { StatColoredWithUnit } from '../../../../../Components/StatDisplay';
...
+        <CardContent sx={{ pt: 1, pb: 1 }}>
           <Box sx={{ display: "flex", gap: 1, alignItems: "center", pb: 1 }}>
             <BootstrapTooltip placement="top" title={<Typography>{t(`slotName.${slotKey}`)}</Typography>}>
               <span>{artifactSlotIcon(slotKey)}</span>
             </BootstrapTooltip>
             <Box flexGrow={1}>
-              <SqBadge color="info">{numSel ? `${numSel} Selected` : `Any`}</SqBadge>
+              <SqBadge color="info">{slotTots[slotKey]}</SqBadge>
             </Box>
-            <Button color="error" size="small" disabled={!mainStatKeys[slotKey].length || disabled} sx={{ mt: -1, mb: -1 }}
```
Comment (nguyentvan7) [inline apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/MainStatSelectionCard.tsx:37]: Why remove the reset button?

Comment (frzyc) [reply apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/MainStatSelectionCard.tsx:37]: the main stat selection is now a "smart toggle", so the reset button is extra clutter. Also, reset sets everything to "off", there is no "off" in smart toggle

**Review by nguyentvan7 — CHANGES_REQUESTED** [review]

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**